### PR TITLE
Port tcdq docs pr

### PR DESCRIPTION
**Pull request by austingmhuang**
**Context:**
This PR refactors the docs for the tcdq module as a whole.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/labs/tcdq/__init__.py [hunk 1]
```diff
@@ -12,11 +12,40 @@
...
+   (MMD) loss, and
+3. optimize the circuit parameters with standard JAX optimizers.
+
+The package supports both qubit circuits and qudit circuits of arbitrary local
+dimension. The public APIs fall into two groups:
+
+1. expectation-value estimators for observables of the circuit output, and
+2. loss functions for matching the circuit distribution to a dataset.
+
+For the mathematical background, we recommend reading `Section 2, Classically Estimating Expectation Values <https://github.com/PennyLaneAI/pennylane/blob/port_tcdq_docs_pr/pennylane/labs/tcdq/notes.md#2-classically-estimating-expectation-values>`_
```
Comment (austingmhuang) [inline pennylane/labs/tcdq/__init__.py]: [BLOCKING] Update links before merging.

File: pennylane/labs/tcdq/__init__.py
Comment (soranjh) [inline pennylane/labs/tcdq/__init__.py:1]: Some high-level comments:

- The init file needs improvement. e.g., the test before the tables is not so useful, you can remove the numerated steps and provide a simple clear descriprion of what this module is about.  You can link the Paper and the notes there too. (when the arxiv link is available)
- In the first example in the init file, please provide comments to specify the steps, e.g., # Define the circuit ...
- The classes have no Arg defined for them. 
- [Non blocking] Some examples are incomplete. For instance, the TrainingOptions is instantiated but not passed to the target function.

File: pennylane/labs/tcdq/__init__.py [hunk 2]
```diff
@@ -113,26 +132,37 @@ def loss_fn(current_params):
...
-of bitstrings), you can use the built-in MMD loss utilities. This integrates
-seamlessly with the ``train`` function using the :class:`~pennylane.labs.tcdq.MMDConfig`
-dataclass.
+To train a circuit so that its output distribution reproduces a target
+dataset of bitstrings, use the built-in Maximum Mean Discrepancy (MMD)
+loss. The MMD is a kernel-based distance between probability distributions.
+Smaller values mean the circuit output is closer to the target data.
+
+The ``bandwidth`` parameter controls how sensitive the loss is to
+fine-grained versus broad differences between distributions. A good
```
Comment (willjmax) [inline pennylane/labs/tcdq/__init__.py:147]: This sentence is confusing.

File: pennylane/labs/tcdq/expval_functions.py [hunk 3]
```diff
@@ -26,61 +28,80 @@
...
+    ...     gates=gates,
+    ...     n_samples=2000,
+    ...     key=jax.random.PRNGKey(42),
+    ...     n_qubits=4,
+    ...     observables=[[3, 3, 0, 0], [0, 0, 3, 3]],  # ZZ on (0,1) and ZZ on (2,3)
+    ... )
+
+    .. seealso::
+
+        `Section 1, Circuit Architecture <https://github.com/PennyLaneAI/pennylane/blob/port_tcdq_docs_pr/pennylane/labs/tcdq/notes.md#1-circuit-architecture>`_
```
Comment (willjmax) [inline pennylane/labs/tcdq/expval_functions.py]: Not clear how this section relates to the circuit config

File: pennylane/labs/tcdq/mmd_loss.py [hunk 4]
```diff
@@ -27,43 +32,77 @@
...
+
+    >>> from pennylane.labs.tcdq import MMDConfig, median_heuristic
+    >>> import numpy as np
+    >>> target_data = np.random.binomial(1, 0.5, size=(200, 6))
+    >>> bw = median_heuristic(target_data)
+    >>> config = MMDConfig(bandwidth=bw, n_ops=64)
+
+    .. seealso::
+
+        `Section 5, Graph-Kernel MMD Loss <https://github.com/PennyLaneAI/pennylane/blob/port_tcdq_docs_pr/pennylane/labs/tcdq/notes.md#5-graph-kernel-mmd-loss>`_
```
Comment (willjmax) [inline pennylane/labs/tcdq/mmd_loss.py]: This section is for qudit MMD loss though?

File: pennylane/labs/tcdq/qudit_expval_functions.py [hunk 5]
```diff
@@ -11,20 +11,21 @@
...
+phase, evaluates a circuit-dependent phase difference, and averages the
+resulting complex integrand.
 
-The public entry point is ``build_qudit_expval_func``. See ``notes.md`` §2 for
-the derivation and Appendix A for the notation-to-code glossary.
+For further information, see
+`Section 2, Classically Estimating Expectation Values <https://github.com/PennyLaneAI/pennylane/blob/port_tcdq_docs_pr/pennylane/labs/tcdq/notes.md#2-classically-estimating-expectation-values>`_,
+`Section 3, General Input States <https://github.com/PennyLaneAI/pennylane/blob/port_tcdq_docs_pr/pennylane/labs/tcdq/notes.md#3-general-input-states>`_,
+and `Section 4, Monte Carlo Statistics <https://github.com/PennyLaneAI/pennylane/blob/port_tcdq_docs_pr/pennylane/labs/tcdq/notes.md#4-monte-carlo-statistics>`_
+of the technical notes.
```
Comment (willjmax) [inline pennylane/labs/tcdq/qudit_expval_functions.py:28]: This should point to the section on the qudit expval function

File: pennylane/labs/tcdq/mmd_loss.py [hunk 6]
```diff
@@ -176,21 +216,57 @@ def mmd_loss(
...
+    >>> import jax.numpy as jnp
+    >>> params = jnp.zeros(len(gates))
+    >>> loss_val = mmd_loss(params, config, mmd_cfg, target)
+    >>> loss_val.shape
+    ()
+
+    .. seealso::
+
+        :func:`~pennylane.labs.tcdq.build_expval_func`,
+        `Section 5, Graph-Kernel MMD Loss <https://github.com/PennyLaneAI/pennylane/blob/port_tcdq_docs_pr/pennylane/labs/tcdq/notes.md#5-graph-kernel-mmd-loss>`_
```
Comment (willjmax) [inline pennylane/labs/tcdq/mmd_loss.py]: This is for qudits, not qubits

File: pennylane/labs/tcdq/training.py [hunk 7]
```diff
@@ -35,57 +43,116 @@
...
     final_params: jnp.ndarray
+    #: Training loss at every optimization step.
     losses: jnp.ndarray
+    #: Validation loss at every step (zeros if disabled).
     val_losses: jnp.ndarray
+    #: Wall-clock time of the training loop in seconds.
     run_time: float
 
 
 class BatchResult(NamedTuple):
```
Comment (willjmax) [inline pennylane/labs/tcdq/training.py:118]: index method is rendering incorrectly

File: pennylane/labs/tcdq/__init__.py [hunk 8]
```diff
@@ -12,11 +12,38 @@
...
+Both qubit and qudit (arbitrary local dimension) circuits are supported.
+
+The module provides:
+
+- **Expectation-value estimators** that compute qubit (Pauli) or
+  qudit (Heisenberg–Weyl) moments of the circuit output.
+- **MMD loss functions** that measure how well the circuit's output
+  distribution matches a target dataset of bitstrings or dit-strings.
+- **Training utilities** that wrap JAX optimizers with convergence
+  checking and validation.
```
Comment (soranjh) [inline pennylane/labs/tcdq/__init__.py]: Remove this.

File: pennylane/labs/tcdq/__init__.py [hunk 9]
```diff
@@ -51,18 +75,8 @@
...
-``pennylane.labs.tcdq`` provides a compact toolkit for constructing and
-simulating phase optimization circuits with JAX. The usual workflow is:
-
-#. Use helpers in :mod:`pennylane.labs.tcdq.utils` to assemble gates and
-   observables.
-#. Configure the circuit with :class:`~pennylane.labs.tcdq.CircuitConfig`.
-#. Build an expectation-value function with
-   :func:`~pennylane.labs.tcdq.build_expval_func` and evaluate it for
-   different parameter sets.
+Example: estimating expectation values
```
Comment (soranjh) [inline pennylane/labs/tcdq/__init__.py]:
```suggestion
The following examples demonstrate several of the key workflows supported by this module.   

Estimating expectation values
```

**Review by soranjh — APPROVED** [review]

File: pennylane/labs/tcdq/__init__.py [hunk 10]
```diff
@@ -12,11 +12,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 r"""
-Phase optimization with JAX: Train Classically Deploy Quantum (TCDQ)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Train Classically, Deploy Quantum (TCDQ).
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (soranjh) [inline pennylane/labs/tcdq/__init__.py]:
```suggestion
This module contains features to enable Train Classical, Deploy Quantum (TCDQ) workflows. 

The supported workflows include fast optimization of instantaneous quantum polynomial (IQP) circuits and their extensions. See [arXiv:2501.04776](https://arxiv.org/abs/2501.04776) and [arXiv:2607.06675](https://arxiv.org/abs/2607.06675) for theoretical details.
```

File: pennylane/labs/tcdq/__init__.py [hunk 11]
```diff
@@ -12,11 +12,38 @@
...
+The module provides:
+
+- **Expectation-value estimators** that compute qubit (Pauli) or
+  qudit (Heisenberg–Weyl) moments of the circuit output.
+- **MMD loss functions** that measure how well the circuit's output
+  distribution matches a target dataset of bitstrings or dit-strings.
+- **Training utilities** that wrap JAX optimizers with convergence
+  checking and validation.
+
+For the mathematical background, see the
```
Comment (soranjh) [inline pennylane/labs/tcdq/__init__.py]: Please make sure this sentence immediately follows the one I added with the paper links.

Comment (austingmhuang) [reply pennylane/labs/tcdq/__init__.py]: Sorry, what should this be? I substituted the links to the md with the links to the paper.

**Review by willjmax — APPROVED** [review]
